tool/gravity/cli/clusterupdate.go

```diff
@@ -96,6 +107,65 @@ func newClusterUpdater(
 	return updater, nil
 }
 
+// checkStatus returns an error if the cluster is degraded.
+// If force is true, warnings will be ignored.
+func checkStatus(ctx context.Context, force bool) error {
+	nodes, err := statusapi.NodeStatuses(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	var failedProbes []string
+	var warningProbes []string
+	for _, node := range nodes {
+		failedProbes = append(failedProbes, node.FailedProbes...)
+		warningProbes = append(warningProbes, node.WarnProbes...)
+	}
+
+	if len(failedProbes) > 0 {
+		return trace.BadParameter("unable to upgrade degraded cluster")
+	}
+
+	if !force && len(warningProbes) > 0 {
+		return trace.BadParameter("cluster has active warnings (use --force to continue)")
+	}
+
+	return nil
+}
+
+// printNodes prints the node statuses
+func printNodes(ctx context.Context, env *localenv.LocalEnvironment) error {
+	clusterOperator, err := env.SiteOperator()
```

Comment:
I don't think we need this extra call to the API to retrieve the status? You already have `status.Agents` field returned by the `FromPlanetAgent` call above which you can just pass to `printAgentStatus()` method to format the message?